src/interface/components/VLIW/modal/VLIWLoadModalComponent.tsx

```diff
@@ -1,147 +1,213 @@
-import * as React from 'react';
-import FileReaderInput from '../../Common/FileReaderInput';
-import { Modal, Button } from 'react-bootstrap';
-import { withTranslation } from 'react-i18next';
-import { connect } from 'react-redux';
-import { toggleLoadModal } from '../../../actions/modals';
-import { bindActionCreators } from 'redux';
-
-import VLIWIntegration from '../../../../integration/vliw-integration';
-import { Code } from '../../../../core/Common/Code';
-import { VLIWCode } from '../../../../core/VLIW/VLIWCode';
-
-export class VLIWLoadModalComponent extends React.Component<any, any> {
-
-    constructor(public props: any, public state: any) {
-        super(props);
-
-        this.close = this.close.bind(this);
-        this.loadCode = this.loadCode.bind(this);
-        this.state = {
-            superescalarCodeError: '',
-            vliwCodeError: ''            
-        }
-    }
+import * as React from "react";
+import FileReaderInput from "../../Common/FileReaderInput";
+import { Modal, Button, Stack, Form, Alert } from "react-bootstrap";
+import { useTranslation, withTranslation } from "react-i18next";
+import { connect } from "react-redux";
+import { toggleLoadModal } from "../../../actions/modals";
+import { bindActionCreators } from "redux";
 
-    close() {
-        this.props.actions.toggleLoadModal(false);
-    };
+import VLIWIntegration from "../../../../integration/vliw-integration";
+import { Code } from "../../../../core/Common/Code";
+import { VLIWCode } from "../../../../core/VLIW/VLIWCode";
+import { useState } from "react";
 
-    handleSuperescalarInputFileChange = (e, results) => {
-        results.forEach(result => {
-            const [e, file] = result;
-            let a = document.getElementById('superescalarCodeInput') as HTMLInputElement;
-            a.value = e.target.result;
-        });
-    }
+const DEFAULT_MODAL_CODE = `
+ADDI	R2 R0 #50
+ADDI	R3 R0 #70
+ADDI	R4 R0 #40
+LF	F0 (R4)
+ADDI	R5 R2 #16
+LOOP:
+LF 	F1 (R2)
+ADDF	F1 F1 F0
+SF	F1 (R3)
+ADDI 	R2 R2 #1
+ADDI	R3 R3 #1
+BNE	R2 R5 LOOP
+`.trim();
```

Comment:
```suggestion
const DEFAULT_MODAL_CODE = `
ADDI	R2 R0 #50
ADDI	R3 R0 #70
ADDI	R4 R0 #40
LF	F0 (R4)
ADDI	R5 R2 #16

LOOP:
LF 	F1 (R2)
ADDF	F1 F1 F0
SF	F1 (R3)
ADDI 	R2 R2 #1
ADDI	R3 R3 #1
BNE	R2 R5 LOOP
`.trim();
```

I'm not sure that `trim()` actually does anything here, I might have left it over in the superscalar loading modal as well :sweat_smile: .

Reply:
Yes, I also think is useless, I just copied your code and I didnt see it

Reply:
Drop them both then, please!